### DISCUSSION TITLE
Update-DbaInstance is failing to extract setup file

### DISCUSSION THREAD

**Discussion by SajalBagchi** [dataplat/dbatools#8917] [Q&A]
I have been upgrading SQL Servers using Update-Dbainstance command from over two years now. All of a sudden, it is failing on multiple servers with the error - "Update failed: Extraction failed with exit code , try specifying a different location using -ExtractPath"

Remote execution failed | Exception calling "Start" with "0" argument(s): "The specified executable is not a valid application for this OS platform."

I am pasting last few lines from Get-Dbatoolslog for details : 

ComputerName : Source_Server_Name
File         : 
FunctionName : Invoke-Command2
Level        : Debug
Line         : 145401
Message      : Found an existing session dbatools_Default_3a57382c-0b48-4c2f-81da-a2198c8d5ab6_Target_SQL_Server_Name, reusing it
ModuleName   : dbatools
Runspace     : 3a57382c-0b48-4c2f-81da-a2198c8d5ab6
Tags         : {}
TargetObject : 
Timestamp    : 5/16/2023 4:18:49 PM
Type         : Debug
Username     : service_account_nm

CallStack    : At Invoke-Command2, <none>: Line 145401
               	At Test-PendingReboot<Process>, <none>: Line 148899
               	At Invoke-DbaAdvancedUpdate, <none>: Line 71520
               	At <ScriptBlock>, <none>: Line 138083
               	At Update-DbaInstance<Process>, <none>: Line 138087
               	At <ScriptBlock>, <none>: Line 5
ComputerName : Source_Server_Name
File         : 
FunctionName : Invoke-Command2
Level        : Debug
Line         : 145401
Message      : Found an existing session dbatools_Default_3a57382c-0b48-4c2f-81da-a2198c8d5ab6_Target_SQL_Server_Name, reusing it
ModuleName   : dbatools
Runspace     : 3a57382c-0b48-4c2f-81da-a2198c8d5ab6
Tags         : {}
TargetObject : 
Timestamp    : 5/16/2023 4:18:49 PM
Type         : Debug
Username     : service_account_nm

CallStack    : At Invoke-DbaAdvancedUpdate, <none>: Line 71536
               	At <ScriptBlock>, <none>: Line 138083
               	At Update-DbaInstance<Process>, <none>: Line 138087
               	At <ScriptBlock>, <none>: Line 5
ComputerName : Source_Server_Name
File         : 
FunctionName : Invoke-DbaAdvancedUpdate
Level        : Debug
Line         : 71536
Message      : Processing Target_SQL_Server_Name with 1 actions
ModuleName   : dbatools
Runspace     : 3a57382c-0b48-4c2f-81da-a2198c8d5ab6
Tags         : {}
TargetObject : 
Timestamp    : 5/16/2023 4:18:49 PM
Type         : Debug
Username     : service_account_nm

CallStack    : At Invoke-DbaAdvancedUpdate, <none>: Line 71575
               	At <ScriptBlock>, <none>: Line 138083
               	At Update-DbaInstance<Process>, <none>: Line 138087
               	At <ScriptBlock>, <none>: Line 5
ComputerName : Source_Server_Name
File         : 
FunctionName : Invoke-DbaAdvancedUpdate
Level        : Verbose
Line         : 71575
Message      : Extracting \\SQLServer\SQLSetup\SQL2019\LatestPatchSetup\SQLServer2019-KB5024276-x64.exe to C:\temp\dbatools_KB5024276_Extract_58a466e74e0944b19d10532ff500a15b
ModuleName   : dbatools
Runspace     : 3a57382c-0b48-4c2f-81da-a2198c8d5ab6
Tags         : {}
TargetObject : 
Timestamp    : 5/16/2023 4:18:49 PM
Type         : Verbose, Debug
Username     : service_account_nm

CallStack    : At Invoke-Program<Process>, <none>: Line 146651
               	At Invoke-DbaAdvancedUpdate, <none>: Line 71576
               	At <ScriptBlock>, <none>: Line 138083
               	At Update-DbaInstance<Process>, <none>: Line 138087
               	At <ScriptBlock>, <none>: Line 5
ComputerName : Source_Server_Name
File         : 
FunctionName : Invoke-Program
Level        : Debug
Line         : 146651
Message      : Acceptable success return codes are [0,3010]
ModuleName   : dbatools
Runspace     : 3a57382c-0b48-4c2f-81da-a2198c8d5ab6
Tags         : {}
TargetObject : 
Timestamp    : 5/16/2023 4:18:49 PM
Type         : Debug
Username     : service_account_nm

CallStack    : At Invoke-Program<Process>, <none>: Line 146654
               	At Invoke-DbaAdvancedUpdate, <none>: Line 71576
               	At <ScriptBlock>, <none>: Line 138083
               	At Update-DbaInstance<Process>, <none>: Line 138087
               	At <ScriptBlock>, <none>: Line 5
ComputerName : Source_Server_Name
File         : 
FunctionName : Invoke-Program
Level        : Verbose
Line         : 146654
Message      : Starting process [\\SQLServer\SQLSetup\SQL2019\LatestPatchSetup\SQLServer2019-KB5024276-x64.exe] with arguments [/x:"C:\temp\dbatools_KB5024276_Extract_58a466e74e0944b19d10532ff500a15b" /quiet] on 
               Target_SQL_Server_Name through Credssp protocol
ModuleName   : dbatools
Runspace     : 3a57382c-0b48-4c2f-81da-a2198c8d5ab6
Tags         : {}
TargetObject : 
Timestamp    : 5/16/2023 4:18:49 PM
Type         : Verbose, Debug
Username     : service_account_nm

CallStack    : At Invoke-Command2, <none>: Line 145401
               	At Invoke-CommandWithFallback, <none>: Line 145456
               	At Invoke-Program<Process>, <none>: Line 146658
               	At Invoke-DbaAdvancedUpdate, <none>: Line 71576
               	At <ScriptBlock>, <none>: Line 138083
               	At Update-DbaInstance<Process>, <none>: Line 138087
               	At <ScriptBlock>, <none>: Line 5
ComputerName : Source_Server_Name
File         : 
FunctionName : Invoke-Command2
Level        : Debug
Line         : 145401
Message      : Found an existing session dbatools_Credssp_3a57382c-0b48-4c2f-81da-a2198c8d5ab6_Target_SQL_Server_Name, reusing it
ModuleName   : dbatools
Runspace     : 3a57382c-0b48-4c2f-81da-a2198c8d5ab6
Tags         : {}
TargetObject : 
Timestamp    : 5/16/2023 4:18:49 PM
Type         : Debug
Username     : service_account_nm

CallStack    : At Stop-Function, <none>: Line 151319
               	At Invoke-Program<Process>, <none>: Line 146663
               	At Invoke-DbaAdvancedUpdate, <none>: Line 71576
               	At <ScriptBlock>, <none>: Line 138083
               	At Update-DbaInstance<Process>, <none>: Line 138087
               	At <ScriptBlock>, <none>: Line 5
ComputerName : Source_Server_Name
File         : 
FunctionName : Invoke-Program
Level        : Warning
Line         : 146663
Message      : Remote execution failed | Exception calling "Start" with "0" argument(s): "The specified executable is not a valid application for this OS platform."
ModuleName   : dbatools
Runspace     : 3a57382c-0b48-4c2f-81da-a2198c8d5ab6
Tags         : {}
TargetObject : 
Timestamp    : 5/16/2023 4:18:49 PM
Type         : Debug, Warning
Username     : service_account_nm

CallStack    : At Stop-Function, <none>: Line 151329
               	At Invoke-DbaAdvancedUpdate, <none>: Line 71580
               	At <ScriptBlock>, <none>: Line 138083
               	At Update-DbaInstance<Process>, <none>: Line 138087
               	At <ScriptBlock>, <none>: Line 5
ComputerName : Source_Server_Name
File         : 
FunctionName : Invoke-DbaAdvancedUpdate
Level        : Warning
Line         : 71580
Message      : Extraction failed with exit code , try specifying a different location using -ExtractPath
ModuleName   : dbatools
Runspace     : 3a57382c-0b48-4c2f-81da-a2198c8d5ab6
Tags         : {}
TargetObject : 
Timestamp    : 5/16/2023 4:18:49 PM
Type         : Debug, Warning
Username     : service_account_nm

CallStack    : At <ScriptBlock>, <none>: Line 137782
               	At Invoke-DbaAdvancedUpdate, <none>: Line 71581
               	At <ScriptBlock>, <none>: Line 138083
               	At Update-DbaInstance<Process>, <none>: Line 138087
               	At <ScriptBlock>, <none>: Line 5
ComputerName : Source_Server_Name
File         : 
FunctionName : Update-DbaInstance
Level        : Warning
Line         : 137782
Message      : Update failed: Extraction failed with exit code , try specifying a different location using -ExtractPath
ModuleName   : dbatools
Runspace     : 3a57382c-0b48-4c2f-81da-a2198c8d5ab6
Tags         : {}
TargetObject : 
Timestamp    : 5/16/2023 4:18:49 PM
Type         : Debug, Warning
Username     : service_account_nm

CallStack    : At Invoke-Command2, <none>: Line 145401
               	At Invoke-CommandWithFallback, <none>: Line 145456
               	At Invoke-DbaAdvancedUpdate, <none>: Line 71616
               	At <ScriptBlock>, <none>: Line 138083
               	At Update-DbaInstance<Process>, <none>: Line 138087
               	At <ScriptBlock>, <none>: Line 5
ComputerName : Source_Server_Name
File         : 
FunctionName : Invoke-Command2
Level        : Debug
Line         : 145401
Message      : Found an existing session dbatools_Credssp_3a57382c-0b48-4c2f-81da-a2198c8d5ab6_Target_SQL_Server_Name, reusing it
ModuleName   : dbatools
Runspace     : 3a57382c-0b48-4c2f-81da-a2198c8d5ab6
Tags         : {}
TargetObject : 
Timestamp    : 5/16/2023 4:18:49 PM
Type         : Debug
Username     : service_account_nm

**Comment by wsmelton** [comment]
What changed?

**Reply by SajalBagchi** [reply]
Not that I am aware of, I have asked windows team to check on all the changes, will post the update, if there is any.

The Patch process run from two different source servers, for each production and non production. Production Patch seems to be running fine and it is only against non production servers which are failing. I have already upgraded dbatool module on source server and have checked the WinRm and connectivity to all the target servers, which is all good.
